timeout while waiting for state to become 'success' (timeout: 2m0s)

#777 attempted to fix this issue but it persists in my environment.

I do not believe this is an issue caused by the rate limit.

Here is the section of the debug log with where the exception in the title occurs:
```  
2023-12-04T16:13:52.730Z [WARN]  provider.terraform-provider-pagerduty_v3.2.2: WaitForState timeout after 2m0s: timestamp=2023-12-04T16:13:52.730Z
2023-12-04T16:13:52.730Z [WARN]  provider.terraform-provider-pagerduty_v3.2.2: WaitForState starting 30s refresh grace period: timestamp=2023-12-04T16:13:52.730Z
2023-12-04T16:13:57.308Z [WARN]  provider.terraform-provider-pagerduty_v3.2.2: WaitForState timeout after 2m0s: timestamp=2023-12-04T16:13:57.308Z
2023-12-04T16:13:57.308Z [WARN]  provider.terraform-provider-pagerduty_v3.2.2: WaitForState starting 30s refresh grace period: timestamp=2023-12-04T16:13:57.308Z
2023-12-04T16:13:57.524Z [WARN]  provider.terraform-provider-pagerduty_v3.2.2: WaitForState timeout after 2m0s: timestamp=2023-12-04T16:13:57.523Z
2023-12-04T16:13:57.524Z [WARN]  provider.terraform-provider-pagerduty_v3.2.2: WaitForState starting 30s refresh grace period: timestamp=2023-12-04T16:13:57.524Z
2023-12-04T16:14:22.732Z [ERROR] provider.terraform-provider-pagerduty_v3.2.2: WaitForState exceeded refresh grace period: timestamp=2023-12-04T16:14:22.731Z
2023-12-04T16:14:22.732Z [ERROR] vertex "module.{pagerduty_service_name}" error: timeout while waiting for state to become 'success' (timeout: 2m0s)
2023-12-04T16:14:22.733Z [ERROR] vertex "module.{pagerduty_service_name} (expand)" error: timeout while waiting for state to become 'success' (timeout: 2m0s)
```

The 200 that occurs right before this indicates the rate limit is not about to be hit:
```
Ratelimit-Limit: 960
Ratelimit-Remaining: 919
Ratelimit-Reset: 58
```

The `WaitForState` messages in the logs makes me think it's related to [an issue upstream in the terraform-plugin-sdk](https://github.com/hashicorp/terraform-plugin-sdk/issues/530). [A fix was submitted for that issue a few years ago](https://github.com/hashicorp/terraform-plugin-sdk/pull/828) but was never reviewed.

See past issues: #765 #760

--- a/pagerduty/config.go
+++ b/pagerduty/config.go
@@ -3,8 +3,11 @@ package pagerduty
 import (
 	"fmt"
 	"log"
+	"net"
 	"net/http"
+	"runtime"
 	"sync"
+	"time"
 
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/logging"
 	"github.com/heimweh/go-pagerduty/pagerduty"
@@ -70,7 +73,18 @@ func (c *Config) Client() (*pagerduty.Client, error) {
 
 	var httpClient *http.Client
 	httpClient = http.DefaultClient
-	httpClient.Transport = logging.NewTransport("PagerDuty", http.DefaultTransport)
+	httpClient.Transport = logging.NewTransport(
+		"PagerDuty",
+		&http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+			DialContext: (&net.Dialer{
+				Timeout:   30 * time.Second,
+				KeepAlive: 30 * time.Second,
+			}).DialContext,
+			IdleConnTimeout:     60 * time.Second,
+			TLSHandshakeTimeout: 10 * time.Second,
+			MaxIdleConnsPerHost: runtime.GOMAXPROCS(0) + 1,
+		})
 
 	var apiUrl = c.ApiUrl
 	if c.ApiUrlOverride != "" {

